packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -145,8 +144,146 @@ class InteractiveViewer extends StatefulWidget {
+  /// ```dart
+  /// import 'package:flutter/widgets.dart';
+  ///
+  /// class Example extends StatelessWidget {
+  ///   const Example({super.key});
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return InteractiveViewer(
+  ///       child: Stack(
+  ///         clipBehavior: Clip.none,
+  ///         children: <Widget>[
+  ///           Positioned(
+  ///             top: -20,
+  ///             left: -20,
+  ///             child: GestureDetector(
+  ///               onTap: () => print('A clicked'),
+  ///               child: const Text('A'),
+  ///             ),
+  ///           ),
+  ///           Positioned(
+  ///             top: 50,
+  ///             left: 20,
+  ///             child: GestureDetector(
+  ///               onTap: () => print('B clicked'),
+  ///               child: const Text('B'),
+  ///             ),
+  ///           ),
+  ///         ],
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// In the above example 'B' can receive hit tests, but 'A' will not register any interactions. This happens
+  /// for [Stack], [CustomMultiChildLayout], or any widget that can position children irregardless
+  /// if the parent is InteractiveViewer.
+  ///
+  /// This is fine for non interactive elements and in most cases having transformChild set to true will be
+  /// the desired behavior.
+  ///
+  /// However if you need hit tests for all children you can set transformChild to false and
+  /// handle child transforms manually.
+  ///
+  /// ```dart
+  /// import 'package:flutter/widgets.dart';
+  /// import 'package:vector_math/vector_math_64.dart' show Matrix4, Quad;
+  ///
+  /// typedef Node = ({Rect rect, Widget child});
+  ///
+  /// class Example extends StatelessWidget {
+  ///   const Example({
+  ///     super.key,
+  ///     required this.controller,
+  ///     required this.nodes,
+  ///   });
+  ///
+  ///   final TransformationController controller;
+  ///   final List<Node> nodes;
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return InteractiveViewer.raw(
+  ///       transformationController: controller,
+  ///       builder: (BuildContext context, Quad viewport) {
+  ///         final Matrix4 matrix = controller.value;
+  ///         return Stack(
+  ///           clipBehavior: Clip.none,
+  ///           children: nodes.map((Node node) {
+  ///             final Rect rect = MatrixUtils.transformRect(matrix, node.rect);
+  ///             return Positioned.fromRect(
+  ///               rect: rect,
+  ///               child: GestureDetector(
+  ///                 onTap: () => print('Node clicked'),
+  ///                 child: node.child,
+  ///               ),
+  ///             );
+  ///           }).toList(),
+  ///         );
+  ///       },
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// Now the child will never exceed the bounds of the parent and always stay static, preserving hit tests.
```

Comment:
Can this phrase be moved above the sample? It feels like it's dangling below now.